extensions/wikia/Listusers/SpecialListusers_helper.php

```diff
@@ -118,71 +119,106 @@ public function loadData() {
 
 			/* initial conditions for SQL query */
 			$where = [
-					'wiki_id' => $this->mCityId,
-					"user_id > 0",
+					self::TABLE . '.wiki_id' => $this->mCityId,
+					self::TABLE . ".user_id > 0",
 					'user_is_closed' => 0
 			];
 
-			/* filter: groups */
-			if ( !empty( $this->mFilterGroup ) && is_array( $this->mFilterGroup ) ) {
-				$whereGroup = array();
-				foreach ( $this->mFilterGroup as $group ) {
-					if ( !empty($group) ) {
-						if ( $group == Listusers::DEF_GROUP_NAME ) {
-							$whereGroup[] = " all_groups = '' ";
-						} else {
-							$whereGroup[] = " all_groups " . $dbs->buildLike( $dbs->anyString(), $group );
-							$whereGroup[] = " all_groups " . $dbs->buildLike( $dbs->anyString(), sprintf("%s;", $group), $dbs->anyString() );
-						}
-					}
-				}
-
-				if ( !empty( $whereGroup ) ) {
-					$where[] = implode( " or ", $whereGroup );
-				}
-			}
-
 			/* filter: user ID  */
 			if ( !empty( $this->mUserId ) ) {
-				$where[] = " user_id = ". intval( $this->mUserId );
+				$where[] = self::TABLE . ".user_id = $this->mUserId";
 			}
 
 			/* filter: number of edits */
 			if ( !empty( $this->mEditsThreshold ) ) {
-				$where[] = " edits >= ". intval( $this->mEditsThreshold );
+				$where[] = "edits >= $this->mEditsThreshold";
 			}
 
-			/* number of records */
-			$data['cnt'] = (int) $dbs->selectField(
-				self::TABLE,
-				'count(*)',
+			/* filter: groups */
+			if ( !empty( $this->mFilterGroup ) && is_array( $this->mFilterGroup ) ) {
+				$filteredGroups = array_filter( $this->mFilterGroup, function ( $group ): bool {
+					return empty( $group ) || $group == Listusers::DEF_GROUP_NAME;
+				} );
+
+				if ( !empty( $filteredGroups ) ) {
+					$where[self::LOCAL_USER_GROUPS_TABLE . '.group_name'] = $this->mFilterGroup;
+				}
+			}
+
+			$result = $dbs->select(
+				[ self::TABLE, self::LOCAL_USER_GROUPS_TABLE ],
+				[ '1' ],
 				$where,
-				__METHOD__ . '::count'
+				__METHOD__,
+				[
+					'GROUP BY' => [
+						self::TABLE . '.wiki_id',
+						self::TABLE . '.user_id',
+					]
+				],
+				[
+					self::LOCAL_USER_GROUPS_TABLE => [
+						'LEFT JOIN',
+						[
+							self::TABLE . '.wiki_id = ' . self::LOCAL_USER_GROUPS_TABLE . '.wiki_id',
+							self::TABLE . '.user_id = ' . self::LOCAL_USER_GROUPS_TABLE . '.user_id',
+							self::LOCAL_USER_GROUPS_TABLE . '.expiry IS NULL OR ' . self::LOCAL_USER_GROUPS_TABLE . '.expiry > NOW()'
+						]
+					]
+				]
 			);
+			// hacky, count returned groups as selectRowCount is not supported
+			$data['cnt'] = count( iterator_to_array( $result ) );
 
 			if ( $data['cnt'] > 0 ) {
 				$userIsBlocked = $user->isBlocked( true, false );
 				$sk = $context->getSkin();
 				/* select records */
 				$oRes = $dbs->select(
-					array( self::TABLE ),
-					array(
-						'user_id',
-						'cnt_groups',
-						'all_groups',
+					[ self::TABLE, self::LOCAL_USER_GROUPS_TABLE, 'lug' => self::LOCAL_USER_GROUPS_TABLE ],
```

Comment:
Wouldn't this be sufficient?
```suggestion
					[ self::TABLE, 'lug' => self::LOCAL_USER_GROUPS_TABLE ],
```

Reply:
Main point here is that we can filter out users without selected groups (using `local_user_groups`) and again join group table (using `local_user_groups as lug`) to fetch all groups for selected users